### PR TITLE
Add a dedicated error variant for planned failures in index scheduler tests

### LINKED ISSUE #3086: Tests: simulated failure within the index scheduler's `process_batch` function uses a confusing error variant
While improving our tests in #3085 I noticed, in some case a failure of milli could « corrupt the task queue ». Or at least that's the error message it returns.

For example for this simple test:
```rust
    #[test]
    fn fail_in_process_batch_for_document_addition() {
        let (index_scheduler, mut handle) =
            IndexScheduler::test(true, vec![(1, FailureLocation::InsideProcessBatch)]);

        let content = r#"
        {
            "id": 1,
            "doggo": "bob"
        }"#;

        let (uuid, mut file) = index_scheduler.create_update_file_with_uuid(0).unwrap();
        let documents_count =
            meilisearch_types::document_formats::read_json(content.as_bytes(), file.as_file_mut())
                .unwrap() as u64;
        file.persist().unwrap();
        index_scheduler
            .register(KindWithContent::DocumentAdditionOrUpdate {
                index_uid: S("doggos"),
                primary_key: Some(S("id")),
                method: ReplaceDocuments,
                content_file: uuid,
                documents_count,
                allow_index_creation: true,
            })
            .unwrap();
        snapshot!(snapshot_index_scheduler(&index_scheduler), name: "registered_the_first_task");
        handle.advance_till([Start, BatchCreated]);

        snapshot!(
            snapshot_index_scheduler(&index_scheduler),
            name: "document_addition_batch_created"
        );

        handle.advance_till([ProcessBatchFailed, AfterProcessing]);
        snapshot!(snapshot_index_scheduler(&index_scheduler), name: "document_addition_failed");
    }
```

The last snapshot look like that:
```
---
source: index-scheduler/src/lib.rs
---
### Autobatching Enabled = true
### Processing Tasks:
[]
----------------------------------------------------------------------
### All Tasks:
0 {uid: 0, status: failed, error: ResponseError { code: 200, message: "Corrupted task queue.", error_code: "internal", error_type: "internal", error_link: "https://docs.meilisearch.com/errors#internal" }, details: { received_documents: 1, indexed_documents: Some(0) }, kind: DocumentAdditionOrUpdate { index_uid: "doggos", primary_key: Some("id"), method: ReplaceDocuments, content_file: 00000000-0000-0000-0000-000000000000, documents_count: 1, allow_index_creation: true }}
----------------------------------------------------------------------
### Status:
enqueued []
failed [0,]
----------------------------------------------------------------------
### Kind:
"documentAdditionOrUpdate" [0,]
----------------------------------------------------------------------
### Index Tasks:
doggos [0,]
----------------------------------------------------------------------
### Index Mapper:
[]
----------------------------------------------------------------------
### Canceled By:

----------------------------------------------------------------------
### Enqueued At:
[timestamp] [0,]
----------------------------------------------------------------------
### Started At:
[timestamp] [0,]
----------------------------------------------------------------------
### Finished At:
[timestamp] [0,]
----------------------------------------------------------------------
### File Store:

----------------------------------------------------------------------
```

------

After further testing, it looks like the task queue isn’t actually corrupted. I can still register some tasks and makes a query on them.
~~But that's still a bug; we should send the one saying that an unexpected panic happened.~~: Read below

### PR DESCRIPTION
**Pull request by Sufflope**
# Pull Request

## Related issue
Fixes #3086

## What does this PR do?
- Add a dedicated test variant in test cfg to avoid reusing a misleading existing error

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by Sufflope** [comment]
Hi!

Wanted to get in touch so I thought I might as well pick a "good first issue".

About the code itself: in the enum error itself, only four existing variants are under the "Irrecoverable errors" comment, but in the `impl ErrorCode for Error` a fifth one is there, the catch-all `anyhow` variant. What should be the truth? I guess the `anyhow` one being a catchall it should be considered irrecoverable too as a baseline, for lack of information. One could imagine later refining that case depending on the underlaying error wrapped by anyhow.

File: index-scheduler/src/error.rs [hunk 1]
```diff
@@ -132,6 +132,24 @@ pub enum Error {
...
+impl Error {
+    pub fn is_recoverable(&self) -> bool {
+        match self {
+            Error::CorruptedTaskQueue
+            | Error::TaskDatabaseUpdate(_)
+            | Error::HeedTransaction(_)
+            | Error::CreateBatch(_) => false,
+            #[cfg(test)]
+            Error::PlannedFailure => false,
+            _ => true,
```
Comment (Kerollmops) [inline index-scheduler/src/error.rs:150]: Could you be explicit on these variants? This way adding a variant to the enum will force us to think about this match.

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Thank you very much for your work 😇

File: index-scheduler/src/error.rs (hunk 1 above)
Comment (Sufflope) [reply index-scheduler/src/error.rs]: Done, I kept the same ordering as in the defining enum but maybe alphabetical ordering would be better in cases one searches a particular value in that long list.

**Comment by Kerollmops** [comment]
bors try

**Comment by Sufflope** [comment]
Rebased on main and added the new error type to the big match

**Review by Kerollmops — COMMENTED** [review]
Thank you very much for the changes @Sufflope 👏
Sorry for the delay. I am merging your PR right now ⛳️

bors merge

**Review by Kerollmops — APPROVED** [review]
bors merge

**Pull request merged**